cdap-api/src/main/java/io/cdap/cdap/api/plugin/InvalidPluginConfigException.java

```diff
@@ -57,4 +57,26 @@ public InvalidPluginConfigException(String message, Set<String> missingPropertie
   public Set<InvalidPluginProperty> getInvalidProperties() {
     return invalidProperties;
   }
+
+  private static String generateErrorMessage(String baseMessage, Set<String> missingProperties,
+                                             Set<InvalidPluginProperty> invalidProperties) {
+    if (missingProperties.isEmpty() && invalidProperties.isEmpty()) {
+      return baseMessage;
+    }
+
+    StringBuilder errorMessage = new StringBuilder(baseMessage);
+    if (missingProperties.size() == 1) {
+      errorMessage.append(String.format(" Required property '%s' is missing.", missingProperties.iterator().next()));
+    } else if (missingProperties.size() > 1) {
+      errorMessage.append(String.format(" Required properties '%s' are missing.",
+                                        String.join(", ", missingProperties)));
+    }
+
+    for (InvalidPluginProperty invalidProperty : invalidProperties) {
+      errorMessage.append(String.format(" '%s' is invalid: %s.",
+                                        invalidProperty.getName(), invalidProperty.getErrorMessage()));
```

Comment:
Also, invalid property is caused by an exception when trying to set the field. So it'll be good if this exception can carry those as suppressed causes.